File: doan2_mobile/lib/models/borrow_record.dart
import 'book.dart';

class BorrowRecord {
  final int? id;
  final Book book;
  final DateTime borrowDate;
  final DateTime dueDate;
  final DateTime? returnDate;
  final String status; // BORROWED, RETURNED, OVERDUE

  BorrowRecord({
    this.id,
    required this.book,
    required this.borrowDate,
    required this.dueDate,
    this.returnDate,
    required this.status,
  });

  factory BorrowRecord.fromJson(Map<String, dynamic> json) {
    return BorrowRecord(
      id: json['id'],
      book: Book.fromJson(json['book']),
      borrowDate: DateTime.parse(json['borrowDate']),
      dueDate: DateTime.parse(json['dueDate']),
      returnDate: json['returnDate'] != null ? DateTime.parse(json['returnDate']) : null,
      status: json['status'] ?? 'BORROWED',
    );
  }
}

File: doan2/src/main/java/com/trsang/doan2/entities/BorrowRecord.java
package com.trsang.doan2.entities;

import jakarta.persistence.*;
import lombok.*;
import java.time.Instant;

@Data
@Builder
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "borrow_records")
public class BorrowRecord {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "book_id", nullable = false)
    private Book book;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "borrow_date", nullable = false, columnDefinition = "TIMESTAMP")
    private Instant borrowDate;

    @Column(name = "due_date", nullable = false, columnDefinition = "TIMESTAMP")
    private Instant dueDate;

    @Column(name = "return_date", columnDefinition = "TIMESTAMP")
    private Instant returnDate;

    @Column(name = "status", nullable = false, length = 20)
    private String status; // BORROWED, RETURNED, OVERDUE
}
